### PR TITLE
fix: module_name may not be valid variable name

### PR DESCRIPTION
**Pull request by plantain-00**
For example: 
```
  "binary": {
    "module_name": "canvas-prebuilt",
    "module_path": "build/Release",
    "host": "https://github.com/node-gfx/node-canvas-prebuilt/releases/download/",
    "remote_path": "v{version}",
    "package_name": "{module_name}-v{version}-{node_abi}-{platform}-{libc}-{arch}.tar.gz"
  }
```
https://github.com/Automattic/node-canvas/blob/master/package.json#L35

### CODE REVIEW

**Review by nicolasnoble — DISMISSED** [review]
Hah. Nice one.

**Comment by springmeyer** [comment]
@plantain-00 this would need some kind of test before merging. Can you add a test?

**Comment by plantain-00** [comment]
The failed CI seems unrelated, I don't know how to fix it:

```txt
not ok 264 Error: Command failed 'V=1 npm install --clang=1 --fallback-to-build=false'
  ---
    operator: error
    at: run (/home/travis/build/mapbox/node-pre-gyp/test/proxy.test.js:316:11)
    stack: |-
      node-pre-gyp ERR! install response status 403 Forbidden on https://mapbox-node-pre-gyp-public-testing-bucket.s3.us-east-1.amazonaws.com/node-pre-gyp/node-pre-gyp-test-app1/v0.1.0/Release/node-v64-linux-x64.tar.gz 
      node-pre-gyp ERR! install error 
      node-pre-gyp ERR! stack Error: response status 403 Forbidden on https://mapbox-node-pre-gyp-public-testing-bucket.s3.us-east-1.amazonaws.com/node-pre-gyp/node-pre-gyp-test-app1/v0.1.0/Release/node-v64-linux-x64.tar.gz
      node-pre-gyp ERR! stack     at fetch.then.res (/home/travis/build/mapbox/node-pre-gyp/lib/install.js:7:1327)
```

**Comment by springmeyer** [comment]
@plantain-00 sorry about the unrelated CI failures. They are discussed at https://github.com/mapbox/node-pre-gyp/pull/491#issuecomment-776223057 and will be addressed next week.

**Comment by springmeyer** [comment]
@plantain-00 I've ask a review question. Also the unrelated CI failures should not be fixed in master. Can you update this branch to latest master and try again?

**Comment by plantain-00** [comment]
the code coverage decrease comes from lib/install.js, but I didn't change the file.

**Comment by bmacnaughton** [comment]
it looks like the code coverage change is because the last run was my internal PR and included the proxy tests. does that make sense @springmeyer?

it's good news that the tests ran without error for an external PR. the proxy tests cannot really test proxying while simulating s3, so they are skipped on external PRs.

**Comment by springmeyer** [comment]
> it looks like the code coverage change is because the last run was my internal PR and included the proxy tests. does that make sense @springmeyer?

seems right. Not sure how we can avoid this problem...

> it's good news that the tests ran without error for an external PR. the proxy tests cannot really test proxying while simulating s3, so they are skipped on external PRs.

Yes, this is great

**Comment by springmeyer** [comment]
@plantain-00 okay I think this PR is ready to merge pending one thing:

 - Can you add docs about this "feature" to the README?

**Comment by springmeyer** [comment]
Thanks @plantain-00!

**Pull request merged**
